keras_cv/layers/preprocessing/channel_shuffle.py

```diff
@@ -0,0 +1,103 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import tensorflow as tf 
+from tensorflow.keras import layers, backend
+
+class ChannelShuffle(layers.Layer):
+    """ChannelShuffle performs shuffling the channel of inputs.  
+
+    Input shape:
+        The expected images should be [0-255] pixel ranges.
+        3D (unbatched) or 4D (batched) tensor with shape:
+        `(..., height, width, channels)`, in `"channels_last"` format
+    Output shape:
+        3D (unbatched) or 4D (batched) tensor with shape:
+        `(..., height, width, channels)`, in `"channels_last"` format
+
+    Args:
+        groups:
+            Number of groups to divide the input channels. Default 3. 
+        seed:
+            Integer. Used to create a random seed.
+
+    Usage:
+    ```python
+    (images, labels), _ = tf.keras.datasets.cifar10.load_data()
+    channelshuffle = keras_cv.layers.preprocessing.ChannelShuffle()
+    augmented_images = channelshuffle(images)
+    ```
+    """
+    def __init__(
+        self,
+        groups=3,
+        seed=None,
+        **kwargs
+    ):
+        super().__init__(**kwargs)
+        self.groups=groups
+        self.seed = seed
+
+    def _channel_shuffling(self, images):
+        unbatched = images.shape.rank == 3
+        
+        if unbatched:
+            images = tf.expand_dims(images, axis=0)
+
+        batch_size, height, width, num_channels = images.get_shape().as_list()
```

Comment:
This is incorrect, it should use `tf.shape`

Reply:
@fchollet 

With this change, one test case failed. 

```python
batch_size = tf.shape(images)[0]
height = tf.shape(images)[1]
width = tf.shape(images)[2]
num_channels = tf.shape(images)[3]

if not num_channels % self.groups == 0:
    raise ValueError("The number of input channels should be divisible by the number of groups."
                             f"Received: channels={num_channels}, groups={self.groups}")
```

```python
def test_in_tf_function(self):
   xs = tf.cast(
       tf.stack([2 * tf.ones((100, 100, 1)), tf.ones((100, 100, 1))], axis=0),
       dtype=tf.float32,
     )

layer = ChannelShuffle(groups=1)

@tf.function
def augment(x):
   return layer(x, training=True)

xs = augment(xs)
```

> ValueError: The number of input channels should be divisible by the number of groups.Received: channels=Tensor("channel_shuffle/strided_slice_3:0", shape=(), dtype=int32), groups=1

Reply:
For the channel number you can actually use `images.shape[3]` since the channel dimension will always be static. However it's important to use `tf.shape` for image dimensions, since they will usually be None for width/height.

Reply:
Fixed.